Modules/Office365DSC/DSCResources/MSFT_SPOUserProfileProperty/MSFT_SPOUserProfileProperty.psm1

```diff
@@ -0,0 +1,250 @@
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $UserName,
+
+        [Parameter()]
+        [Microsoft.Management.Infrastructure.CimInstance[]]
+        $Properties,
+
+        [Parameter()]
+        [ValidateSet("Present")]
+        [System.String]
+        $Ensure = "Present",
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount
+    )
+
+    Write-Verbose -Message "Getting SPO Profile Properties for user {$UserName}"
+
+    Test-MSCloudLogin -O365Credential $GlobalAdminAccount `
+                      -Platform PnP
+
+    $nullReturn = @{
+        UserName           = $UserName
+        Properties         = $Properties
+        GlobalAdminAccount = $GlobalAdminAccount
+        Ensure             = 'Absent'
+    }
+
+    try
+    {
+        $currentProperties = Get-PnPUserProfileProperty -Account $UserName
+        $currentProperties = $currentProperties.UserProfileProperties
+
+        $propertiesValue = @()
+
+        foreach ($key in $currentProperties.Keys)
+        {
+            $convertedProperty = Get-SPOUserProfilePropertyInstance -Key $key -Value $currentProperties[$key]
+            $propertiesValue += $convertedProperty
+        }
+        $result =  @{
+            UserName           = $UserName
+            Properties         = $propertiesValue
+            GlobalAdminAccount = $GlobalAdminAccount
+            Ensure             = "Present"
+        }
+
+        Write-Verbose -Message "Get-TargetResource Result: `n $(Convert-O365DscHashtableToString -Hashtable $result)"
+        return $result
+    }
+    catch
+    {
+        throw $_
+        return $nullReturn
+    }
+
+}
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $UserName,
+
+        [Parameter()]
+        [Microsoft.Management.Infrastructure.CimInstance[]]
+        $Properties,
+
+        [Parameter()]
+        [ValidateSet("Present")]
+        [System.String]
+        $Ensure = "Present",
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount
+    )
+
+    Write-Verbose -Message "Setting Profile Properties for user {$UserName}"
+
+    Test-MSCloudLogin -O365Credential $GlobalAdminAccount `
+                      -Platform PnP
+
+    foreach ($property in $Properties)
+    {
+        Write-Verbose "Setting Profile Property {$($property.Key)} as {$($property.Value)}"
+        Set-PnPUserProfileProperty -Account $UserName -PropertyName $property.Key -Value $property.Value
```

Comment:
Do you really want to set all properties here, even the ones that didn't change? This will overwrite already existing values and might trigger an unwanted action somewhere in O365 (crawl for example) because of an update.

Reply:
Fixed by only updating delta properties that are different.